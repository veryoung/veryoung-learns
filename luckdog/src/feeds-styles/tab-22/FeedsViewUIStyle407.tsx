// 选择阅读喜好
import React from 'react';
import { View, Text, Image, StyleSheet } from '@tencent/hippy-react-qb';
import FeedsViewItem from '../FeedsViewItem';
import FeedsViewContainer from '../common/FeedsViewContainer';
import FeedsProtect from '../../mixins/FeedsProtect';
import FeedIcon from '../../framework/FeedsIcon';
import { reportUDS, strictExposeReporter, BusiKey } from '@/luckdog';
import { FeedsUIStyle } from '../../framework/FeedsConst';
import { FeedsTheme } from './components/utils';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';
import { CommonProps } from '../../entity';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginHorizontal: 12,
    marginBottom: 0,
    marginTop: FeedsUIStyle.FEEDS_CARD_MARGIN_VERTICAL,
    height: 52,
    borderRadius: 8,
    backgroundColors: FeedsTheme.SkinColor.N2_1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  tagTitle: {
    fontSize: FeedsUIStyle.T2,
    colors: FeedsTheme.SkinColor.N1,
    alignSelf: 'center',
    paddingVertical: 16,
  },
  iconView: {
    width: 12,
    height: 12,
    alignSelf: 'center',
  },
});

interface Props extends CommonProps {
  index: number;
}

@FeedsProtect.protect
export default class FeedsViewUIStyle407 extends FeedsViewItem<Props> {
  public static getRowType() {
    return 407;
  }

  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'FeedsViewUIStyle407');
  }

  public handleClick = (url) => {
    reportUDS(BusiKey.CLICK__CARD, this.props);
    super.onClick(url);
  };

  public onLayout = (event) => {
    strictExposeReporter.addExpoItem({
      cardIndex: this.props.index,
      rect: event.layout,
      forceCheckExpo: true,
    });
  };

  public render() {
    countReRender(this, 'FeedsViewUIStyle407');
    const { itemBean = {} } = this.props;
    const { title, url } = itemBean;
    if (!title || !url) return null;
    return (
      <FeedsViewContainer
        onLayout={this.onLayout}
        parentProps={this.props}
        noPressState
        noPadding
      >
        <View
          style={styles.container}
          onClick={() => this.handleClick(url)}
        >
          <Text style={styles.tagTitle}>{title}</Text>
          <Image
            style={styles.iconView}
            source={{ uri: FeedIcon.arrow }}
            noPicMode={{ enable: false }}
            tintColors={styles.tagTitle.colors}
            reportData={{ sourceFrom: itemBean.item_id }}
          />
        </View>
      </FeedsViewContainer>
    );
  }
}
