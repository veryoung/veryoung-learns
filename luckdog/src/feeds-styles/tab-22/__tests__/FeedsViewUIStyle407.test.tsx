import React from 'react';
import renderer from 'react-test-renderer';

import FeedsViewUIStyle407 from '../FeedsViewUIStyle407';
import { commonProps } from '@test/mocks/props';

describe('<FeedsViewUIStyle407/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        index: 0,
      };
      const tree = renderer.create(<FeedsViewUIStyle407 {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
