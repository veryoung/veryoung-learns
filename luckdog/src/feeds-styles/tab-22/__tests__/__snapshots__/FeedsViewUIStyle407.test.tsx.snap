// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle407/> render testing render normal 1`] = `null`;
